chart-parts/cluster-roles.yaml

```diff
@@ -0,0 +1,136 @@
+rules:
+- apiGroups: ['admissionregistration.k8s.io']
+  resources: ['mutatingwebhookconfigurations']
+  verbs:     ['*']
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-webhook" .Release.Namespace) }}
+  labels:
+    {{- template "scf.role-labels" "eirini-webhook" }}
+    {{- template "scf.labels" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-webhook" .Release.Namespace) }}
+subjects:
+- kind: ServiceAccount
+  name: eirini
+  namespace: {{.Release.Namespace}}
+
+# eirini-csr cluster role for secret-generator service account.
+# Used to create certificate signing requests for the opi-registry certificate.
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-csr" .Release.Namespace) }}
+  labels:
+    {{- template "scf.role-labels" "eirini-csr" }}
+    {{- template "scf.labels" . }}
+rules:
+- apiGroups: [certificates.k8s.io]
+  resources: [certificatesigningrequests]
+  verbs:     [create, get, delete]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-csr" .Release.Namespace) }}
+  labels:
+    {{- template "scf.role-labels" "eirini-csr" }}
+    {{- template "scf.labels" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-csr" .Release.Namespace) }}
+subjects:
+- kind: ServiceAccount
+  name: secret-generator
+  namespace: {{ .Release.Namespace }}
+
+{{- if eq (toString .Values.env.KUBE_CSR_AUTO_APPROVAL) "true" }}
+
+# eirini-csr-approver cluster role for secret-generator service account.
+# Used to auto-aaprove certificate signing requests for the opi-registry certificate.
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-csr-approver" .Release.Namespace) }}
+  labels:
+    {{- template "scf.role-labels" "eirini-csr-approver" }}
+    {{- template "scf.labels" . }}
+rules:
+- apiGroups: [certificates.k8s.io]
+  resources: [certificatesigningrequests/approval]
+  verbs:     [update]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-csr-approver" .Release.Namespace) }}
+  labels:
+    {{- template "scf.role-labels" "eirini-csr-approver" }}
+    {{- template "scf.labels" . }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ template "fissile.SanitizeName" (printf "%s-eirini-csr-approver" .Release.Namespace) }}
+subjects:
+- kind: ServiceAccount
+  name: secret-generator
+  namespace: {{ .Release.Namespace }}
+{{- end }}
+
+{{- else }}{{/* if .Values.enable.eirini */}}
```

Comment:
I wonder if it would make things easier to read if we had separate Eirini and Diego cluster role files. (This does not block merging, of course)

Reply:
We could; I'm kind of indifferent, as `diego` is kind of `not-eirini`, but I can make them separate files, if you prefer.

Reply:
Meh, I'm not too worried about it. Hopefully we'll just drop diego at some point :p